xarray/plot/plot.py

```diff
@@ -151,6 +179,55 @@ def plot(darray, row=None, col=None, col_wrap=None, ax=None, rtol=0.01,
     return plotfunc(darray, **kwargs)
 
 
+def _infer_line_data(darray, x, y, hue):
+    error_msg = ('must be either None or one of ({0:s})'
+                 .format(', '.join([repr(dd) for dd in darray.dims])))
+    ndims = len(darray.dims)
+
+    if x is not None and x not in darray.dims:
+        raise ValueError('x ' + error_msg)
+
+    if y is not None and y not in darray.dims:
+        raise ValueError('y ' + error_msg)
+
+    if x is not None and y is not None:
+        raise ValueError('You cannot specify both x and y kwargs'
+                         'for line plots.')
+
+    if ndims == 1:
```

Comment:
This will need to be changed to follow #2151. One of us should rebase on the other's branch,. How do you think we should proceed?

Reply:
The question is whether it's likely that there will more commits in #2151 or not. If not I can rebase and continue from there. If yes, it's probably better the other way around.

(As said above, it's my first contribution to open source software stuff so I don't really know how this works and what would be simpler)

Reply:
I ended up merging #2151 first, so you should you merge the latest updates into this PR, e.g.,
```
git checkout master
git pull upstream master
git checkout line_FacetGrid
git merge master
```

Reply:
OK, I'm pretty sure I messed this up but I might have fixed it too. All this rebasing is new to me. Sorry. Was it normal that after I did all that I got a list of many many modified files that I had to commit?